<a href="https://colab.research.google.com/github/riyazahmadshaik/computationalmethods-5731/blob/master/Copy_of_INFO5731_Assignment_four.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **INFO5731 Assignment Four**

In this assignment, you are required to conduct topic modeling, sentiment analysis, and regression analysis.

# **Question 1: Topic Modeling**

(30 points). This question is designed to help you understand topic modeling better as well as how to visualize topic modeling results, aims to collect the human meanings of documents. Based on the yelp review data (only the review text will be used for this question), which can be download from Dropbox: https://www.dropbox.com/s/59hsrk56sfwh9u2/Assignment%20four%20data%20Yelp%20%28question%201%20and%202%29.zip?dl=0, **select two models** and write a python program to **identify the top 20 topics (with 15 words for each topic) in the dataset**. Before answering this question, please review the materials in lesson 8, as well as the introduction of these models by the links provided.

(1)   Labeled LDA (LLDA): https://github.com/JoeZJH/Labeled-LDA-Python

(2)   Biterm Topic Model (BTM): https://github.com/markoarnauto/biterm

(3)   HMM-LDA: https://github.com/dongwookim-ml/python-topic-model

(4)   SupervisedLDA: https://github.com/dongwookim-ml/python-topic-model/tree/master/notebook

(5)   Relational Topic Model: https://github.com/dongwookim-ml/python-topic-model/tree/master/notebook

(6)   LDA2VEC: https://github.com/cemoody/lda2vec

(7)   BERTopic: https://github.com/MaartenGr/BERTopic

(8)   LDA+BERT Topic Modeling: https://www.kaggle.com/dskswu/topic-modeling-bert-lda

(9)   Clustering for Topic models: (paper: https://arxiv.org/abs/2004.14914), (code: https://github.com/adalmia96/Cluster-Analysis)


**The following information should be reported:**

(1) Top 20 clusters for topic modeling.

(2) Summarize and describe the topic for each cluster. 

(3) Visualize the topic modeling reasults by using pyLDAVis: https://www.machinelearningplus.com/nlp/topic-modeling-visualization-how-to-present-results-lda-models/#14.-pyLDAVis


In [ ]:
# Write your code here


import pandas as pd
papers = pd.read_csv("Assignment4_File_5731.csv",encoding = "ISO-8859-1")
papers_data = papers['Clean_Title']
papers.head()

Document_ID                                        Clean_Title Sentiment
0            1  viewer actual went tiff wit film didnt want be...  Positive
1            2  outstand movi haunt perform best charact devel...  Positive
2            3                                certain peopl relat   Neutral
3            4                               perfect everi aspect  Positive
4            5                                          hype real  Positive

In [ ]:

import gensim
from gensim.utils import simple_preprocess
def sent_to_words(sentences):
    for sentence in sentences:
        yield(gensim.utils.simple_preprocess(str(sentence), deacc=True))  # deacc=True removes punctuations
data = papers.Clean_Title.values.tolist()
data_words = list(sent_to_words(data))
print(data_words)

[['viewer', 'actual', 'went', 'tiff', 'wit', 'film', 'didnt', 'want', 'believ', 'hype', 'absolut', 'masterpiec', 'phoenix', 'certifi', 'legend'], ['outstand', 'movi', 'haunt', 'perform', 'best', 'charact', 'develop', 'ever', 'seen'], ['certain', 'peopl', 'relat'], ['perfect', 'everi', 'aspect'], ['hype', 'real'], ['masterpiec'], ['amaz', 'movi', 'exist'], ['went', 'second', 'time', 'watch'], ['psycholog', 'studi', 'rather', 'superhero', 'flick'], ['joaquin', 'oscar', 'joker', 'best', 'dark', 'suspens', 'thriller', 'darker', 'dark', 'knight'], ['venic', 'review'], ['represint', 'real', 'life', 'joker'], ['final', 'real', 'movi'], ['good', 'lord'], ['spoon', 'feed', 'cgi', 'fuel', 'faux', 'drama'], ['oscar', 'phoenix'], ['critic', 'useless'], ['joker', 'endgam'], ['would', 'call', 'masterpiec'], ['one', 'best', 'act', 'perform', 'ive', 'ever', 'seen'], ['dont', 'forget', 'smile'], ['ok', 'film'], ['extrem', 'overr'], ['believ', 'hype'], ['masterpiec'], ['great', 'dont', 'know'], ['that',

In [ ]:

bigram = gensim.models.Phrases(data_words, min_count=5, threshold=100) # higher threshold fewer phrases.
trigram = gensim.models.Phrases(bigram[data_words], threshold=100)
bigram_mod = gensim.models.phrases.Phraser(bigram)
trigram_mod = gensim.models.phrases.Phraser(trigram)

/usr/local/lib/python3.6/dist-packages/gensim/models/phrases.py:598: UserWarning: For a faster implementation, use the gensim.models.phrases.Phraser class
  warnings.warn("For a faster implementation, use the gensim.models.phrases.Phraser class")


In [ ]:
import nltk
nltk.download('stopwords')
from nltk.corpus import stopwords
stop_words = stopwords.words('english')
stop_words.extend(['from', 'subject', 're', 'edu', 'use'])
def remove_stopwords(texts):
    return [[word for word in simple_preprocess(str(doc)) if word not in stop_words] for doc in texts]
def make_bigrams(texts):
    return [bigram_mod[doc] for doc in texts]
def make_trigrams(texts):
    return [trigram_mod[bigram_mod[doc]] for doc in texts]
def lemmatization(texts, allowed_postags=['NOUN', 'ADJ', 'VERB', 'ADV']):
    """https://spacy.io/api/annotation"""
    texts_out = []
    for sent in texts:
        doc = nlp(" ".join(sent)) 
        texts_out.append([token.lemma_ for token in doc if token.pos_ in allowed_postags])
    return texts_out

[nltk_data] Downloading package stopwords to /root/nltk_data...
[nltk_data]   Unzipping corpora/stopwords.zip.


In [ ]:
import spacy
data_words_nostops = remove_stopwords(data_words)
# Form Bigrams
data_words_bigrams = make_bigrams(data_words_nostops)
nlp = spacy.load("en_core_web_sm", disable=['parser', 'ner'])
data_lemmatized = lemmatization(data_words_bigrams, allowed_postags=['NOUN', 'ADJ', 'VERB', 'ADV'])
print(data_lemmatized)

[['go', 'want', 'believ', 'hype'], ['perform', 'good', 'charact', 'develop', 'ever', 'see'], ['certain', 'peopl', 'relat'], ['perfect', 'everi', 'aspect'], ['hype'], [], ['exist'], ['go', 'second', 'time', 'watch'], ['rather', 'superhero'], ['good', 'dark', 'suspen'], ['review'], ['real', 'life', 'joker'], ['final', 'real', 'movi'], ['good'], ['feed', 'cgi', 'fuel', 'faux', 'drama'], [], ['critic', 'useless'], [], ['would', 'call'], ['good', 'act', 'perform', 'have', 'ever', 'see'], ['forget', 'smile'], ['film'], ['extrem', 'overr'], [], [], ['great', 'know'], ['life'], ['extrem', 'overr'], ['requiem', 'broken', 'man'], ['rise'], ['brilliant', 'best', 'joker', 'have', 'ever', 'see'], [], ['old', 'peopl', 'get', 'stupid'], [], ['probabl', 'good', 'comic', 'book', 'perform', 'make', 'speechless', 'end', 'venic'], ['perform', 'have', 'come', 'know'], ['see', 'tonight', 'make', 'account', 'tell', 'everyon', 'good'], [], [], ['anyon', 'rate', 'cinema'], [], ['best', 'movi', 'ever'], [], ['a

In [ ]:
import gensim.corpora as corpora
id2word = corpora.Dictionary(data_lemmatized)
texts = data_lemmatized
corpus = [id2word.doc2bow(text) for text in texts]
print(corpus)

[[(0, 1), (1, 1), (2, 1), (3, 1)], [(4, 1), (5, 1), (6, 1), (7, 1), (8, 1), (9, 1)], [(10, 1), (11, 1), (12, 1)], [(13, 1), (14, 1), (15, 1)], [(2, 1)], [], [(16, 1)], [(1, 1), (17, 1), (18, 1), (19, 1)], [(20, 1), (21, 1)], [(7, 1), (22, 1), (23, 1)], [(24, 1)], [(25, 1), (26, 1), (27, 1)], [(27, 1), (28, 1), (29, 1)], [(7, 1)], [(30, 1), (31, 1), (32, 1), (33, 1), (34, 1)], [], [(35, 1), (36, 1)], [], [(37, 1), (38, 1)], [(6, 1), (7, 1), (8, 1), (9, 1), (39, 1), (40, 1)], [(41, 1), (42, 1)], [(43, 1)], [(44, 1), (45, 1)], [], [], [(46, 1), (47, 1)], [(26, 1)], [(44, 1), (45, 1)], [(48, 1), (49, 1), (50, 1)], [(51, 1)], [(6, 1), (9, 1), (25, 1), (40, 1), (52, 1), (53, 1)], [], [(11, 1), (54, 1), (55, 1), (56, 1)], [], [(7, 1), (8, 1), (57, 1), (58, 1), (59, 1), (60, 1), (61, 1), (62, 1), (63, 1)], [(8, 1), (40, 1), (47, 1), (64, 1)], [(7, 1), (9, 1), (60, 1), (65, 1), (66, 1), (67, 1), (68, 1)], [], [], [(69, 1), (70, 1), (71, 1)], [], [(6, 1), (29, 1), (52, 1)], [], [(72, 1)], [(66, 

In [ ]:
from pprint import pprint
lda_model = gensim.models.LdaMulticore(corpus=corpus,
                                       id2word=id2word,
                                       num_topics=20, 
                                       random_state=100,
                                       chunksize=100,
                                       passes=20,
                                       per_word_topics=True)
pprint(lda_model.print_topics(num_words=4))
doc_lda = lda_model[corpus]

[(0, '0.490*"good" + 0.073*"thing" + 0.073*"astonish" + 0.003*"far"'),
 (1, '0.125*"good" + 0.064*"first" + 0.064*"act" + 0.064*"see"'),
 (2, '0.143*"rather" + 0.143*"superhero" + 0.007*"brain" + 0.007*"rate"'),
 (3, '0.154*"bad" + 0.079*"suspen" + 0.079*"hour" + 0.079*"good"'),
 (4, '0.143*"extrem" + 0.143*"overr" + 0.073*"sooner" + 0.073*"rate"'),
 (5, '0.125*"joke" + 0.064*"broken" + 0.064*"requiem" + 0.064*"peopl"'),
 (6, '0.126*"everyon" + 0.126*"cinemat" + 0.126*"seem" + 0.006*"rate"'),
 (7, '0.165*"review" + 0.008*"astonish" + 0.008*"feel" + 0.008*"rate"'),
 (8, '0.112*"actor" + 0.112*"final" + 0.112*"real" + 0.112*"movi"'),
 (9, '0.247*"hype" + 0.166*"believ" + 0.085*"want" + 0.085*"go"'),
 (10, '0.199*"real" + 0.133*"life" + 0.068*"movi" + 0.068*"joker"'),
 (11, '0.102*"good" + 0.101*"book" + 0.101*"sheep" + 0.101*"comic"'),
 (12, '0.073*"feed" + 0.073*"faux" + 0.073*"fuel" + 0.073*"would"'),
 (13, '0.101*"exist" + 0.101*"ten" + 0.101*"asleep" + 0.101*"fall"'),
 (14, '0.112*"e

1.Features used for Topic Modelling
1.Tokenizing the sentences into list of words and removing the stopwords, unnecesarry characters if any.
2.Gensim's Phrases Model can implement the bigrams,trigrams,quadgrams and more.The two important argument for the phrases are min_count and threshold.If the threshold is higher we get fewer phrases.
3.The two main inputs of the LDA topic model are dictionary and the corpus.Gensim creates a unique id for each word in the document.The produced corpus shown above is a mapping of(word_id,word_frequency).
4.LDA MODEL:After creating the dictionary and corpus,we need to provide the num_topics as well.
I used chucksize and passes while building LDA model using gensims.Chunksize decides how many documents are processed at a time in the training algorithm and passes decides how often we can train the model on the entire corpus.
5.The above LDA model is build with the top 10 topics where each topic is the combination of keywords and each keyword contributes a certain weightage to the topic.
lda_model.print_topics(num_words=4) is used to print the keywords of each topic and the importance(weightage) of each keyword with 4 topic terms for each topic.

Top 20 Clusters for topic Modeling
The top 20 clusters obtained are:
[(0, '0.490*"good" + 0.073*"thing" + 0.073*"astonish" + 0.003*"far"'),
 (1, '0.125*"good" + 0.064*"first" + 0.064*"act" + 0.064*"see"'),
 (2, '0.143*"rather" + 0.143*"superhero" + 0.007*"brain" + 0.007*"rate"'),
 (3, '0.154*"bad" + 0.079*"suspen" + 0.079*"hour" + 0.079*"good"'),
 (4, '0.143*"extrem" + 0.143*"overr" + 0.073*"sooner" + 0.073*"rate"'),
 (5, '0.125*"joke" + 0.064*"broken" + 0.064*"requiem" + 0.064*"peopl"'),
 (6, '0.126*"everyon" + 0.126*"cinemat" + 0.126*"seem" + 0.006*"rate"'),
 (7, '0.165*"review" + 0.008*"astonish" + 0.008*"feel" + 0.008*"rate"'),
 (8, '0.112*"actor" + 0.112*"final" + 0.112*"real" + 0.112*"movi"'),
 (9, '0.247*"hype" + 0.166*"believ" + 0.085*"want" + 0.085*"go"'),
 (10, '0.199*"real" + 0.133*"life" + 0.068*"movi" + 0.068*"joker"'),
 (11, '0.102*"good" + 0.101*"book" + 0.101*"sheep" + 0.101*"comic"'),
 (12, '0.073*"feed" + 0.073*"faux" + 0.073*"fuel" + 0.073*"would"'),
 (13, '0.101*"exist" + 0.101*"ten" + 0.101*"asleep" + 0.101*"fall"'),
 (14, '0.112*"everyon" + 0.112*"feel" + 0.112*"brain" + 0.112*"wash"'),
 (15, '0.084*"realli" + 0.043*"understand" + 0.043*"go" + 0.043*"perfect"'),
 (16, '0.070*"see" + 0.070*"make" + 0.070*"account" + 0.036*"tonight"'),
 (17, '0.085*"watch" + 0.085*"forget" + 0.085*"smile" + 0.085*"time"'),
 (18, '0.134*"film" + 0.068*"ever" + 0.068*"see" + 0.068*"good"'),
 (19, '0.061*"peopl" + 0.061*"cheap" + 0.061*"perform" + 0.061*"know"')]


Summarize and describe the topics for each cluster
0.Joker is a real hype with every perfect aspect.
1.Spend good time and best movie making.
2.Everyone feel like broken and brain wash.
3.Bad film and insult to best comic book.
4.Psychological study rather than a superhero.
5.Over extreme performance.
6.Saw tonight and made an account to tell everyone about faux drama.
7.One act performance of people I have ever seen.
8.Believe the hype,Good film to watch.
9.Went blind dint enjoy that life of joker really disappointed.
10.Just amazing. How does this movie exist.
11.I feel like everyone is brain washed.
12.Only certain people can relate.
13.Kind of a letdown.
14.Best dc movie since the dark knight Rises
15.Joaquin oscar joker is best Dark suspense thriller. Darker than dark knight.
16.Stop comparing it with Endgame
17.Perfect in every aspect.
18.I don't get it
19.The hype is real

# **Question 2: Yelp Review Sentiment Analysis**

(30 points). Sentiment analysis also known as opinion mining is a sub field within Natural Language Processing (NLP) that builds machine learning algorithms to classify a text according to the sentimental polarities of opinions it contains, e.g., positive, negative, neutral. The purpose of this question is to develop a machine learning classifier for sentiment analysis. Based on the dataset from assignment three, write a python program to implement a sentiment classifier and evaluate its performance. Notice: **80% data for training and 20% data for testing**.

The data can be download from Dropbox: https://www.dropbox.com/s/59hsrk56sfwh9u2/Assignment%20four%20data%20Yelp%20%28question%201%20and%202%29.zip?dl=0 

The data was saved in json format, here is an example of the data (for this task, you only need to use the star rating and the review text fields):

{
    // string, 22 character unique review id
    "review_id": "zdSx_SD6obEhz9VrW9uAWA",

    // string, 22 character unique user id, maps to the user in user.json
    "user_id": "Ha3iJu77CxlrFm-vQRs_8g",

    // string, 22 character business id, maps to business in business.json
    "business_id": "tnhfDv5Il8EaGSXZGiuQGg",

    // integer, star rating
    "stars": 4,

    // string, date formatted YYYY-MM-DD
    "date": "2016-03-09",

    // string, the review itself
    "text": "Great place to hang out after work: the prices are decent, and the ambience is fun. It's a bit loud, but very lively. The staff is friendly, and the food is good. They have a good selection of drinks.",

    // integer, number of useful votes received
    "useful": 0,

    // integer, number of funny votes received
    "funny": 0,

    // integer, number of cool votes received
    "cool": 0
}

The sentiment of can be accessed based on the star rating, if no star information avaliable for a record, just remove that record. Detail star and sentiment level can be matched blew:

Very positive = 5 stars

Positive = 4 stars

Neutral = 3 stars

Negative = 2 stars

Very negative = 1 star

Here is code for yelp data preprocessing: https://github.com/Yelp/dataset-examples. 

Answer the following questions:

(1) Features used for sentiment classification and explain why you select these features (tf-idf, sentiment lexicon, word2vec, etc). Considering achieve the best performance as you can. 

(2) Select two of the supervised learning algorithm from scikit-learn library: https://scikit-learn.org/stable/supervised_learning.html#supervised-learning, to build a sentiment classifier respectively. 

(3) Compare the performance over accuracy, precision, recall, and F1 score for the two algorithms you selected. Here is the reference of how to calculate these metrics: https://towardsdatascience.com/accuracy-precision-recall-or-f1-331fb37c5cb9. 

In [ ]:
# Write your code here

import pandas as pd
import warnings
warnings.filterwarnings("ignore")
df = pd.read_csv('Assignment4_File_5731.csv',encoding = "ISO-8859-1")
X = df['Clean_Title']
y = df['Sentiment']



In [ ]:
#LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import CountVectorizer
from sklearn import metrics
from sklearn.metrics import accuracy_score
from sklearn.metrics import precision_score
from sklearn.metrics import recall_score
from sklearn.metrics import f1_score
from sklearn.linear_model import LogisticRegression

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state=42)

vect = CountVectorizer(min_df=2, ngram_range=(1, 2))
X_train = vect.fit(X_train).transform(X_train) 
X_test = vect.transform(X_test)
logreg = LogisticRegression()
logreg.fit(X_train,y_train)
y_pred=logreg.predict(X_test)
print("Accuracy:",metrics.accuracy_score(y_test, y_pred))
print("Precision:",metrics.precision_score(y_test, y_pred,average = 'macro'))
print("Recall:",metrics.recall_score(y_test, y_pred,average = 'macro'))
print("f1 score:",metrics.f1_score(y_test, y_pred,average = 'macro'))

Accuracy: 0.7
Precision: 0.47222222222222215
Recall: 0.48888888888888893
f1 score: 0.47771836007130125


In [ ]:
#support Vector Machine
from sklearn import svm
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state=42)
vect = CountVectorizer(min_df=2, ngram_range=(1, 2))
X_train = vect.fit(X_train).transform(X_train) 
X_test = vect.transform(X_test)
clf = svm.SVC(kernel='linear')
clf.fit(X_train, y_train)
y_pred = clf.predict(X_test)
print("Accuracy:",metrics.accuracy_score(y_test, y_pred))
print("Precision:",metrics.precision_score(y_test, y_pred,average = 'macro'))
print("Recall:",metrics.recall_score(y_test, y_pred,average = 'macro'))
print("f1 score:",metrics.f1_score(y_test, y_pred,average = 'macro'))

Accuracy: 0.5
Precision: 0.3790849673202614
Recall: 0.36296296296296293
f1 score: 0.30769230769230765


Features used for Sentiment Classification A feature or aspect is an attribute of entity.Different features can generate different sentiment analysis. The features used for sentiment classification are clean titles of the joker movie reviews and sentiment i.e. if the title is positive,negative or neutral.I chose these features to obtain the better accuarcy because the accuracy depends on the relatinship between the features.

For feature selection we need to divide the dataset into dependent and independent variables.Here dependent is the sentiment columns and independent is the clean title from our dataset.

Some of the techniques used for feature selection are Mutual Information,Chi-Square,Information gain and TF-idf to select geatures from high dimentionality of feature set.

1.LogisticRegression Logistic Regression is one of the most simple and commonly used Machine Learning algorithms for two-class classification.It describes and estimates the relationship between one dependent binary variable and independent variables.
1.I have dividied the columns into two types variables dependent(y) and independent variable(X). 2.splitting the dataset into 80% training data and 20% test data using function train_test_split().In this function we need to pass three parameters dependent variable,independent variable and test_size and random_state can be used to select the records randomly. 3.LogisticRegression() function is used to create a Logistic Regression classifier object. 4.Fit() function is used to fit the model on the train set and predict() function is used to make predictions on the test set. 5.Accurcay,Precision,recall and f1 score are calculated to evaluate the metrics.
2.Support Vector Machine
SVM constructs a hyperplane in multidimensional space to separate different classes. SVM generates optimal hyperplane in an iterative manner, which is used to minimize an error. The core idea of SVM is to find a maximum marginal hyperplane(MMH) that best divides the dataset into classes. 1.I have dividied the columns into two types variables dependent(y) and independent variable(X). 2.splitting the dataset into 80% training data and 20% test data using function train_test_split().In this function we need to pass three parameters dependent variable,independent variable and test_size and random_state can be used to select the records randomly. 3.SVC() function is used to create a Support Vector classifier object by passing an argument kernel as the linear kernel.A linear Kernel is nothing but a dot product of any two given observations. 4.Fit() function is used to fit the model on the train set and predict() function is used to make predictions on the test set. 5.Accurcay,Precision,recall and f1 score are calculated to evaluate the metrics.

Comparing the algorithms based on accuracy,precision,recall amd f1 score

I have chosen the logistic regression and support vector machine algorithms because our dataset is very small and for accurate results over a small dataset we have few algorithms like logistic regression, SVM using linear, Naive Bayes, RandomForest.

From the baove results based on the accuracy,precision,recall and f1 score we can say that logistic regression classifier is best suited for this dataset as the accuracy is 70% which is good enough and the f1 score is close to 0.5 which means the algorithm is good for this dataset.

# **Question 3: House price prediction**

(40 points). You are required to build a **regression** model to predict the house price with 79 explanatory variables describing (almost) every aspect of residential homes. The purpose of this question is to practice regression analysis, an supervised learning model. The training data, testing data, and data description files can be download from Dropbox: https://www.dropbox.com/s/52j9hpxppfo921o/assignment4-question3-data.zip?dl=0. Here is an axample for the implementation: https://towardsdatascience.com/linear-regression-in-python-predict-the-bay-areas-home-price-5c91c8378878. 


In [ ]:
import numpy as np
import pandas as pd
from sklearn.impute import SimpleImputer
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import cross_val_score

train = pd.read_csv('train.csv')
test = pd.read_csv('test.csv')
train_X = train.drop('SalePrice', axis=1)
train_y = train.SalePrice
test_X = test
onehot_train_X = pd.get_dummies(train_X)
onehot_test_X = pd.get_dummies(test_X)
train_X, test_X = onehot_train_X.align(onehot_test_X, join='left', axis=1)
my_imputer = SimpleImputer()
train_X = my_imputer.fit_transform(train_X)
test_X = my_imputer.transform(test_X)
reg = LinearRegression()
cv_scores = cross_val_score(reg, train_X, train_y, cv=5)
reg = LinearRegression()
reg.fit(train_X, train_y)
predictions = reg.predict(test_X)
my_submission = pd.DataFrame({'Id': test.Id, 'SalePrice':predictions})
my_submission.to_csv("Predicted_House_Price.csv")
my_submission

Id      SalePrice
0     1461  112709.903813
1     1462  161010.363863
2     1463  186961.179807
3     1464  197300.132022
4     1465  205541.292943
...    ...            ...
1454  2915   86095.528874
1455  2916   81610.279372
1456  2917  181080.339264
1457  2918  117021.339522
1458  2919  225002.291367

[1459 rows x 2 columns]

In [ ]:

#Performance evaluation for the baove model.
reg = LinearRegression()
reg.fit(train_X,train_y)
X_train , X_test, Y_train, Y_test = train_test_split(
    train_X,train_y,
        test_size=0.20,
        random_state=42)
predictions = reg.predict(X_test)
print("Mean Absolute Error", metrics.mean_absolute_error(Y_test, predictions))
print("Mean Square Error",metrics.mean_squared_error(Y_test, predictions))
print("R2 Score",metrics.r2_score(Y_test, predictions))

Mean Absolute Error 14426.740961107349
Mean Square Error 522409274.01144296
R2 Score 0.931892169915545
